### PR TITLE
Create CreateCategoryPage

### PR DESCRIPTION
**Pull request by luacmartins**
### Details
Adds the ability to create categories

### Fixed Issues
$ https://github.com/Expensify/App/issues/35710

### Tests
1. As a Collect policy admin, navigate to `Settings > Workspace > Categories`
2. Verify that you can see the `Add category` button at the top of the screen
3. Press the button and verify that the RHP opens up with a text input for `Category name`
4. Press `Save` without any name and verify that you see an error
5. Enter a category name and press `Save`
6. Verify that the category is created and you can see it in the list

- [x] Verify that no errors appear in the JS console

### Offline tests
1. As a Collect policy admin, navigate to `Settings > Workspace > Categories`
2. Go offline
4. Press the `Add category` button
6. Enter a category name and press `Save`
7. Verify that the category is created and you can see it in the list with 50% opacity
8. Go back online
9. Verify that the category is no longer with 50% opacity

### QA Steps
Same as tests

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/22219519/587cd31c-cacc-424b-a3c5-b3b6a3594776

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/22219519/3e954de4-d8e2-4d17-9657-2b197a0fa021

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/22219519/7cabd3df-8828-42ff-8bba-9b07a67ea9f6

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/22219519/46ab1141-7773-4271-a9bf-8689019b610d

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/22219519/737520e3-00c0-4e28-9691-22d3dfc8bfc7

</details>

### CODE REVIEW

**Comment by luacmartins** [comment]
@Expensify/design

File: src/CONST.ts [hunk 1]
```diff
@@ -1640,6 +1640,7 @@ const CONST = {
     FORM_CHARACTER_LIMIT: 50,
     LEGAL_NAMES_CHARACTER_LIMIT: 150,
     LOGIN_CHARACTER_LIMIT: 254,
+    CATEGORY_NAME_LIMIT: 256,
```
Comment (mananjadhav) [inline src/CONST.ts:1643]: Unaware about the limits on each feature, hence a question, do we document the limit somewhere?  If you see L1645, we have a generic `TITLE_CHARACTER_LIMIT: 100`, which I thought would be used here?

Comment (luacmartins) [reply src/CONST.ts:1643]: The API validates these differently. I don't think there's documentation other than the CONSTS defined in the API. I checked this value against the API and 256 is correct for category names.

**Comment by shawnborton** [comment]
When we land in the "Add category" page (RHP), I think the input should auto-focus:

CleanShot 2024-03-05 at 13 38 40@2x

**Comment by shawnborton** [comment]
Why doesn't more of the page scroll? I think I would expect almost everything besides the page header to scroll here... perhaps even the buttons should be scrolling. 

CleanShot 2024-03-05 at 13 39 27@2x

**Comment by luacmartins** [comment]
Added autofocus. 

>Why doesn't more of the page scroll? I think I would expect almost everything besides the page header to scroll here... perhaps even the buttons should be scrolling.

Not sure if we ever discussed what's scrollable here. I'd say that we should tackle that as a separate issue since the scope of this PR is to add the `Create category` functionality and we're not changing the current scrolling behavior we have on main. I can create an issue for it if you'd like.

**Comment by shawnborton** [comment]
Yes please! Curious what the other designers think about scrolling too. But right now it feels like way too much of the page is locked.

File: src/pages/workspace/categories/CreateCategoryPage.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            if (!ValidationUtils.isRequiredFulfilled(categoryName)) {
+                errors.categoryName = 'workspace.categories.categoryRequiredError';
+            } else if (policyCategories?.[categoryName]) {
+                errors.categoryName = 'workspace.categories.existingCategoryError';
+            } else if (categoryName === CONST.INVALID_CATEGORY_NAME) {
+                errors.categoryName = 'workspace.categories.invalidCategoryName';
+            } else if ([...categoryName].length > CONST.CATEGORY_NAME_LIMIT) {
+                // Uses the spread syntax to count the number of Unicode code points instead of the number of UTF-16
+                // code units.
+                ErrorUtils.addErrorMessage(errors, 'categoryName', ['common.error.characterLimitExceedCounter', {length: [...categoryName].length, limit: CONST.TITLE_CHARACTER_LIMIT}]);
```
Comment (mananjadhav) [inline src/pages/workspace/categories/CreateCategoryPage.tsx:52]:
```suggestion
                ErrorUtils.addErrorMessage(errors, 'categoryName', ['common.error.characterLimitExceedCounter', {length: [...categoryName].length, limit: CONST.CATEGORY_NAME_LIMIT}]);
```

We're using `category_name_limit` in the condition and `title_character_limit` in the error message.

File: src/pages/workspace/categories/CreateCategoryPage.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                    <FormProvider
+                        formID={ONYXKEYS.FORMS.WORKSPACE_CATEGORY_CREATE_FORM}
+                        onSubmit={createCategory}
+                        submitButtonText={translate('common.save')}
+                        validate={validate}
+                        style={[styles.mh5, styles.flex1]}
+                    >
+                        <InputWrapper
+                            InputComponent={TextInput}
+                            maxLength={100}
```
Comment (mananjadhav) [inline src/pages/workspace/categories/CreateCategoryPage.tsx:87]: Here too we set the max limit as 256 based on CONST, but hardcoded as 100 here.

**Comment by mananjadhav** [comment]
@luacmartins So we don't have an offline support for adding the categories right? Save button is disabled when we go offline.

**Comment by luacmartins** [comment]
@mananjadhav it should be available. I just pushed a fix. Thanks for catching that.

File: src/pages/workspace/categories/CreateCategoryPage.tsx (hunk 2 above)
Comment (luacmartins) [reply src/pages/workspace/categories/CreateCategoryPage.tsx]: Updated

File: src/pages/workspace/categories/CreateCategoryPage.tsx (hunk 3 above)
Comment (luacmartins) [reply src/pages/workspace/categories/CreateCategoryPage.tsx]: Updated

**Comment by dannymcclain** [comment]
>Curious what the other designers think about scrolling too. But right now it feels like way too much of the page is locked.

Same, but I agree we should handle it for all of these types of pages. @luacmartins can you just tag `@Expensify/design` on the issue you create?

**Comment by mananjadhav** [comment]
> The validation doesn't work. I think the `[....].length` isn't working.

You can ignore this. I think it's because of `maxLength` it truncated the input.

**Comment by luacmartins** [comment]
Created issue for the scroll behavior [here](https://github.com/Expensify/App/issues/37782)

**Comment by mananjadhav** [comment]
Reloading the new category route, takes you to the last chat and doesn't retain the category list in the background. Other pages such as Members Invite, etc. are retained to their respective workspace pages.


https://github.com/Expensify/App/assets/3069065/33d62bee-7913-49ee-ac25-d1d2037ca443

File: src/CONST.ts [hunk 4]
```diff
@@ -1720,6 +1721,7 @@ const CONST = {
     MAX_64BIT_LEFT_PART: 92233,
     MAX_64BIT_MIDDLE_PART: 7203685,
     MAX_64BIT_RIGHT_PART: 4775807,
+    INVALID_CATEGORY_NAME: '###',
```
Comment (mananjadhav) [inline src/CONST.ts:1724]: Can you help me for my knowledge we don't allow `###` (3 #) but allow the other counts?

File: src/ONYXKEYS.ts [hunk 5]
```diff
@@ -326,6 +326,8 @@ const ONYXKEYS = {
         ADD_DEBIT_CARD_FORM: 'addDebitCardForm',
         ADD_DEBIT_CARD_FORM_DRAFT: 'addDebitCardFormDraft',
         WORKSPACE_SETTINGS_FORM: 'workspaceSettingsForm',
+        WORKSPACE_CATEGORY_CREATE_FORM: 'workspaceCategoryCreate',
+        WORKSPACE_CATEGORY_CREATE_FORM_DRAFT: 'workspaceCategoryCreateDraft',
```
Comment (mananjadhav) [inline src/ONYXKEYS.ts:330]: Is it necessary to create a draft form key even we don't have draft option?

File: src/CONST.ts (hunk 4 above)
Comment (luacmartins) [reply src/CONST.ts:1724]: I'm not too sure haha This is a 9 year old check we have in the API 🤷

File: src/ONYXKEYS.ts (hunk 5 above)
Comment (luacmartins) [reply src/ONYXKEYS.ts:330]: Yea, I found that weird too, but type checks fail without it.

Comment (mananjadhav) [reply src/ONYXKEYS.ts:330]: Cool.

File: src/CONST.ts (hunk 4 above)
Comment (mananjadhav) [reply src/CONST.ts:1730]: Okay, let's leave it then ;)

**Comment by mananjadhav** [comment]
> Reloading the new category route, takes you to the last chat and doesn't retain the category list in the background. Other pages such as Members Invite, etc. are retained to their respective workspace pages.

Last comment @luacmartins.

**Comment by luacmartins** [comment]
@mananjadhav fixed the reload issue

**Comment by mananjadhav** [comment]
@luacmartins We now have a few Typescript errors.

**Comment by luacmartins** [comment]
Pushed a fix already

**Comment by mananjadhav** [comment]
Thanks I think most looks good. I've covered all cases. I couldn't check the length error without removing the `maxLength`. Hence I tested by removing the property temporarily.

**Comment by luacmartins** [comment]
Nice! Thanks for the thorough review!

**Comment by mananjadhav** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/3069065/fde6bcc1-03c7-4947-b601-fed15d32bb0a

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/3069065/192c7eb7-ae17-4fbc-a38d-b151ae86ee6c

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/3069065/2a9b64c0-e617-4d07-96b2-3693b5b84c2f

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/3069065/1416e16b-aae6-4945-a62f-3821cd5fd4db

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/3069065/942f5e9b-16c8-40a2-8dfb-89f63e2d78fe

https://github.com/Expensify/App/assets/3069065/df3af3bb-289a-4514-b152-4ce3922f401b

https://github.com/Expensify/App/assets/3069065/9c6b7a77-c99f-4eb2-9891-f46ad08da25f

web-create-category-3

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/3069065/3aeb3e4a-83a5-4a3f-9894-76878e0804bf

</details>

**Review by mananjadhav — APPROVED** [review]

**Comment by mananjadhav** [comment]
Thanks for the quick responses. I am hoping we've covered everything. :v:

File: src/pages/workspace/categories/CreateCategoryPage.tsx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+                // code units.
+                ErrorUtils.addErrorMessage(errors, 'categoryName', ['common.error.characterLimitExceedCounter', {length: [...categoryName].length, limit: CONST.CATEGORY_NAME_LIMIT}]);
+            }
+
+            return errors;
+        },
+        [policyCategories],
+    );
+
+    const createCategory = (values: FormOnyxValues<typeof ONYXKEYS.FORMS.WORKSPACE_CATEGORY_CREATE_FORM>) => {
```
Comment (youssef-lr) [inline src/pages/workspace/categories/CreateCategoryPage.tsx:60]: Why not use `useCallback` here as well?

**Comment by youssef-lr** [comment]
@luacmartins a minor polish item, new categories are always added to the bottom and are not sorted alphabetically. If you log out then log back in they get sorted.

https://github.com/Expensify/App/assets/9680864/3fc8d9b2-0445-4a3d-a1bc-9a7326cb0b49

**Comment by luacmartins** [comment]
@youssef-lr we have a new pattern that will add the cateegory, scroll and highlight it, but that will be implemented as part of https://github.com/Expensify/App/issues/35717 for members and then we'll use the same solution for categories

File: src/pages/workspace/categories/CreateCategoryPage.tsx (hunk 6 above)
Comment (luacmartins) [reply src/pages/workspace/categories/CreateCategoryPage.tsx:59]: added!

**Comment by luacmartins** [comment]
@youssef-lr updated!

File: src/pages/workspace/categories/CreateCategoryPage.tsx (hunk 6 above)
Comment (youssef-lr) [reply src/pages/workspace/categories/CreateCategoryPage.tsx:59]: I don't think you pushed this change 😅

**Comment by youssef-lr** [comment]
>we have a new pattern that will add the cateegory, scroll and highlight it

Oh so it's intentional that we'll not sort them and just add it to the bottom? even though after a refresh it will move?

**Comment by youssef-lr** [comment]
@luacmartins FWIW I think it's a simple fix to just add `.sort((a, b) => localeCompare(a.name, b.name) as String)` [here](https://github.com/Expensify/App/blob/3dfd5129f27b4db8e655ca721330a387f89528f0/src/pages/workspace/categories/WorkspaceCategoriesPage.tsx#L71) if we still want them sorted (if the list is long, we can still scroll to their position while they're still sorted and not necessarily in the bottom, later on when we implement that pattern). I won't block on this though.

Screenshot 2024-03-06 at 18 28 14

**Comment by luacmartins** [comment]
Ah lol I never pushed the change 😓  Ok pushed the change with useCallback and your sorting suggestion too

**Review by youssef-lr — DISMISSED** [review]
LGTM, thanks for the changes!

**Review by youssef-lr — DISMISSED** [review]

**Review by youssef-lr — APPROVED** [review]

**Pull request merged**